app/kubemci/pkg/validations/validations.go

```diff
@@ -89,3 +91,69 @@ func nodePortSameInAllClusters(backend v1beta1.IngressBackend, namespace string,
 	}
 	return nil
 }
+
+func ServerVersionsNewEnough(clients map[string]kubeclient.Interface) (bool, error) {
+	for key := range clients {
+		glog.Infof("Checking client %s", key)
+		discoveryClient := clients[key].Discovery()
+		if discoveryClient == nil {
+			return false, fmt.Errorf("no discovery client in client: %+v", clients[key])
+		}
+		ver, err := discoveryClient.ServerVersion()
+		if err != nil {
+			return false, fmt.Errorf("could not get discovery client to lookup server version: %s", err)
+		}
+		glog.Infof("ServerVersion: %+v", ver)
+		newEnough, err := serverVersionNewEnough(ver.GitVersion)
+		if err != nil {
+			return false, fmt.Errorf("error checking server version: %s", err)
+		}
+		if !newEnough {
+			fmt.Println("Cluster", key, "is not running a supported kubernetes version. Need at least 1.8.1.")
+			return false, nil
+		}
+
+	}
+	return true, nil
+}
+
+func serverVersionNewEnough(version string) (bool, error) {
+	// Example string we're matching: v1.9.3-gke.0
+	re := regexp.MustCompile("^v([0-9]*).([0-9]*).([0-9]*)")
+	matches := re.FindStringSubmatch(version)
+	glog.V(4).Infof("version string matches: %v\n", matches)
+
+	if len(matches) < 3 {
+		return false, fmt.Errorf("did not find 3 components to version number %s", version)
+	}
+	// Major version
+	major, err := strconv.ParseUint(matches[1], 10 /*base*/, 32 /*bitSize*/)
+	if err != nil {
+		return false, err
+	}
+	if major > 1 {
+		return true, nil
+	} else if major < 1 {
+		return false, nil
+	}
+	//Minor version
+	minor, err := strconv.ParseUint(matches[2], 10 /*base*/, 32 /*bitSize*/)
+	if err != nil {
+		return false, err
+	}
+	if minor > 8 {
+		return true, nil
+	} else if minor < 8 {
+		return false, nil
+	}
+	// Patch version
+	patch, err := strconv.ParseUint(matches[3], 10 /*bases*/, 32 /*bitSize*/)
+	if err != nil {
+		return false, err
+	}
+	glog.V(2).Infof("Got version: major:", major, "minor:", minor, "patch:", patch)
+	if patch >= 1 {
+		return true, nil
+	}
+	return false, nil
```

Comment:
Also check !1.10.0? Or you want to do that in a separate PR?

Reply:
Done. And it was much easier after factoring it out into a "parse version number" function.